sqlglot/optimizer/pushdown_projections.py

```diff
@@ -61,6 +61,9 @@ def pushdown_projections(expression, schema=None, remove_unused_selections=True)
             if remove_unused_selections:
                 _remove_unused_selections(scope, parent_selections, schema)
 
+            if scope.expression.is_star:
+                continue
+
```

Comment:
We don't want to hit the following section where we update `referenced_columns` if there's a star projection in the outer scope, because then we might treat the inner scope's projections as "unused", even though this isn't safe.

For example:

```python
>>> from sqlglot.optimizer.pushdown_projections import pushdown_projections
>>> import sqlglot
>>> pushdown_projections(sqlglot.parse_one("SELECT * FROM ((SELECT c FROM t1) JOIN t2)")).sql()
'SELECT * FROM ((SELECT 1 AS _ FROM t1), t2)'
```

Reply:
I also tried to include a check like `scope.parent.expression.is_star` in the above condition:

https://github.com/tobymao/sqlglot/blob/726306e9a77a700555b31fbe136e6438f711178f/sqlglot/optimizer/pushdown_projections.py#L42-L45

But this broke e.g. the following test:

```sql
SELECT a FROM (SELECT * FROM aa UNION ALL SELECT * FROM bb UNION ALL SELECT * from cc);
SELECT _q_0.a AS a FROM (SELECT aa.a AS a FROM aa AS aa UNION ALL SELECT bb.a AS a FROM bb AS bb UNION ALL SELECT cc.a AS a FROM cc AS cc) AS _q_0;
```

Because we weren't pushing down the projection `a` into the inner scope, due to the fact that `UNION ALL` scope is the parent of `SELECT * FROM aa` and it's a star.